From 3ce059e780e1b687198873c20f7d13cb10ea6634 Mon Sep 17 00:00:00 2001
From: Wu Zhangjin <zhangjin.wu@windriver.com>
Date: Tue, 21 Sep 2010 17:02:15 +0800
Subject: [PATCH 8/8] tracing/ftrace: MIPS: Fix potential NULL pointer reference

Before, there is no NULL pointer passed to
ftrace_arch_code_modify_post_process(), it works fine, but after the
following two patches are merged in:

	ftrace: pass NULL to frace_arch_code_modify_post_proces_ar
	grsec: Integrate grsecurity and PaX to CGL kernel

there will be a NULL pointer reference, so, fix it.

Signed-off-by: Wu Zhangjin <zhangjin.wu@windriver.com>
---
 arch/mips/kernel/ftrace.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/arch/mips/kernel/ftrace.c b/arch/mips/kernel/ftrace.c
index 785965b..b69c707 100644
--- a/arch/mips/kernel/ftrace.c
+++ b/arch/mips/kernel/ftrace.c
@@ -83,7 +83,12 @@ static inline void flush_icache_from(unsigned long start)
 
 int ftrace_arch_code_modify_post_process(void *data)
 {
-	int *command = data;
+	int *command;
+
+	if (data == NULL)
+		return 0;
+
+	command = data;
 
 	if (*command & (FTRACE_ENABLE_CALLS | FTRACE_DISABLE_CALLS)) {
 		__flush_cache_all();
-- 
1.7.2.1

